config/api/src/main/java/org/jboss/windup/config/tags/Tag.java

```diff
@@ -0,0 +1,98 @@
+package org.jboss.windup.config.tags;
+
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import org.apache.commons.collections4.set.UnmodifiableSet;
+
+/**
+ * Represents a tag. Determined by it's lowercased name.
+ */
+public final class Tag
+{
+    private final String name;
+    /**
+     * Keeps the "contains" relation.
+     */
+    private final Set<Tag> containedTags = new HashSet<>();
+    private final Set<Tag> containingTags = new HashSet<>();
```

Comment:
what about calling this as childrenTags and parentTags? Is it possible to have multiple parent tags as well?